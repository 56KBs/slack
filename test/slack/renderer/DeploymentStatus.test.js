const { DeploymentStatus } = require('../../../lib/slack/renderer/deployment-status');
const deploymentStatusPendingFixture = require('../../fixtures/webhooks/deployment/status_pending.json');
const deploymentStatusSuccessFixture = require('../../fixtures/webhooks/deployment/status_success.json');
const deploymentStatusSuccessMasterFixture = require('../../fixtures/webhooks/deployment/status_success_master.json');
const branches = require('../../fixtures/branches.json');

describe('Deployment status rendering', () => {
  describe('(attachment)', () => {
    test('works for pending status', () => {
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus: deploymentStatusPendingFixture.deployment_status,
        deployment: deploymentStatusPendingFixture.deployment,
        repository: deploymentStatusPendingFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getAttachment()).toMatchSnapshot();
    });

    test('works for success status', () => {
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus: deploymentStatusSuccessFixture.deployment_status,
        deployment: deploymentStatusSuccessFixture.deployment,
        repository: deploymentStatusSuccessFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getAttachment()).toMatchSnapshot();
    });
    test('works for failure status', () => {
      const deploymentStatus = {
        ...deploymentStatusSuccessFixture.deployment_status,
        state: 'failure',
      };
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus,
        deployment: deploymentStatusSuccessFixture.deployment,
        repository: deploymentStatusSuccessFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getAttachment()).toMatchSnapshot();
    });
    test('works for error status', () => {
      const deploymentStatus = {
        ...deploymentStatusSuccessFixture.deployment_status,
        state: 'error',
      };
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus,
        deployment: deploymentStatusSuccessFixture.deployment,
        repository: deploymentStatusSuccessFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getAttachment()).toMatchSnapshot();
    });
    test('works for deployments on default branch', () => {
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus: deploymentStatusSuccessMasterFixture.deployment_status,
        deployment: deploymentStatusSuccessMasterFixture.deployment,
        repository: deploymentStatusSuccessMasterFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getAttachment()).toMatchSnapshot();
    });
  });
  describe('(full message)', () => {
    test('works', () => {
      const deploymentStatusMessage = new DeploymentStatus({
        deploymentStatus: deploymentStatusPendingFixture.deployment_status,
        deployment: deploymentStatusPendingFixture.deployment,
        repository: deploymentStatusPendingFixture.repository,
        branches,
      });
      expect(deploymentStatusMessage.getRenderedMessage()).toMatchSnapshot();
    });
  });
});
