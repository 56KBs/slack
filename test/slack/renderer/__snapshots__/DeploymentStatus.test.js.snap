// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deployment status rendering (attachment) works for deployments on default branch 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#28a745",
  "fallback": "Successfully deployed master to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Successfully deployed <https://github.com/github-slack/app/commit/115d3a3e80e3e85af270c982a33d4b31bf934342|\`master\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
}
`;

exports[`Deployment status rendering (attachment) works for error status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#cb2431",
  "fallback": "Error when deploying deployments to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Error when deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`deployments\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
}
`;

exports[`Deployment status rendering (attachment) works for failure status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#cb2431",
  "fallback": "Deploying deployments to github-slack-app failed",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`deployments\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app> failed",
}
`;

exports[`Deployment status rendering (attachment) works for pending status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#dbab09",
  "fallback": "Deploying deployments to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`deployments\`> to <https://dashboard.heroku.com/apps/github-slack-app|github-slack-app>",
}
`;

exports[`Deployment status rendering (attachment) works for success status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#28a745",
  "fallback": "Successfully deployed deployments to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`deployments\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
}
`;

exports[`Deployment status rendering (full message) works 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#dbab09",
      "fallback": "Deploying deployments to github-slack-app",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`deployments\`> to <https://dashboard.heroku.com/apps/github-slack-app|github-slack-app>",
    },
  ],
}
`;
